### PR TITLE
Bump to tree-sitter-0.3 and tree-sitter-python 0.4.

### PR DESCRIPTION
**Pull request by patrickt**

### CODE REVIEW

File: src/Language/TypeScript/Assignment.hs [hunk 1]
```diff
@@ -456,7 +451,7 @@ callSignatureParts = contextualize' <$> Assignment.manyThrough comment (postCont
       Nothing -> (typeParams, formalParams, annotation)
 
 callSignature :: Assignment Term
-callSignature =  makeTerm <$> symbol Grammar.CallSignature <*> children (TypeScript.Syntax.CallSignature <$> (fromMaybe <$> emptyTerm <*> optional (term typeParameters)) <*> formalParameters <*> (fromMaybe <$> emptyTerm <*> optional (term typeAnnotation')))
+callSignature =  makeTerm <$> location <*> children (TypeScript.Syntax.CallSignature <$> (fromMaybe <$> emptyTerm <*> optional (term typeParameters)) <*> formalParameters <*> (fromMaybe <$> emptyTerm <*> optional (term typeAnnotation')))
```
Comment (robrix) [inline src/Language/TypeScript/Assignment.hs]: There’s really no specific symbol we should be looking for when constructing this? `children` not preceded by `symbol` is almost always a mistake.

Comment (patrickt) [reply src/Language/TypeScript/Assignment.hs]: It possibly could be the wrong thing. The relevant clause was renamed from `$.call_signature` to `$._call_signature` and its children appear to be invoked directly:

```diff
   (ambient_declaration
     (internal_module
-      (identifier)
-      (statement_block
+      name: (identifier)
+      body: (statement_block
         (function_signature
-          (identifier)
-          (call_signature
-            (formal_parameters (required_parameter (identifier) (type_annotation (predefined_type))))
-            (type_annotation (predefined_type))))
-        (lexical_declaration (variable_declarator (identifier) (type_annotation (predefined_type))))
+          name: (identifier)
+          parameters: (formal_parameters (required_parameter (identifier) (type_annotation (predefined_type))))
+          return_type: (type_annotation (predefined_type)))
+        (lexical_declaration (variable_declarator
+          name: (identifier)
+          type: (type_annotation (predefined_type))))
```

We’ll see if tests fail.

Comment (robrix) [reply src/Language/TypeScript/Assignment.hs]: I think we should 🔥 the `children` call here, like we did in the other similar cases.

File: test/Generators.hs [hunk 2]
```diff
@@ -10,13 +9,12 @@ module Generators
...
 import qualified Hedgehog.Gen as Gen
 import qualified Data.Source
-import Data.Functor.Identity
 import Data.Scientific (Scientific)
 import Data.Ratio ((%))
 import qualified Data.Scientific as Scientific
 
 
-source :: (GenBase m ~ Identity, MonadGen m) => Hedgehog.Range Int -> m Data.Source.Source
+source :: MonadGen m => Hedgehog.Range Int -> m Data.Source.Source
```
Comment (robrix) [inline test/Generators.hs]: 🙇

File: src/Language/TypeScript/Assignment.hs (hunk 1 above)
Comment (robrix) [reply src/Language/TypeScript/Assignment.hs]:
```suggestion
callSignature =  makeTerm <$> location <*> (TypeScript.Syntax.CallSignature <$> (fromMaybe <$> emptyTerm <*> optional (term typeParameters)) <*> formalParameters <*> (fromMaybe <$> emptyTerm <*> optional (term typeAnnotation')))
```

Tho we should also probably make sure that this isn’t being used in `expressionChoices` in that case.

**Pull request merged**